code/glTF2AssetWriter.inl

```diff
@@ -397,17 +454,24 @@ namespace glTF2 {
 
     inline void Write(Value& obj, Sampler& b, AssetWriter& w)
```

Comment:
Because all of the properties of a `Sampler` are optional, it's possible that a totally empty `Sampler` may be written to the file, which is valid glTF, but a better solution might be to omit the sample altogether.

This is a very common case: exporting an OBJ with textures as glTF2 will result in empty `Sampler`s.

Reply:
Yep. I wanted to do something like that, but didn't spend too much time on it this weekend. I couldn't figure out an easy way to check if the Sampler resource was empty, and prevent if from being written: There's a general AssetWriter::WriteObjects method which delegates to each Type's Write method. But then I'm not sure how to cleanly check If the object is empty and stop it from being written. Further, I'm not sure how to prevent texture from referencing that now-deleted Sampler.

Or is there a simpler solution?

Reply:
The empty samplers are created in https://github.com/Matter-and-Form/assimp/blob/feature/gltf2-importer/code/glTF2Exporter.cpp#L243. So, you should be able to check if at least one of the sampler parameters is non-default, and only add it to the texture in that case, right?

Reply:
True. At that point (in the Exporter), it's not actually empty, but has default values set. It's only in AssetWriter that those default values get pruned.